### ISSUE TITLE
Gizmo appearing on pivot, not actual mesh center

### ISSUE THREAD

**Issue by neuro-tech** [HiddenMonk/Unity3DRuntimeTransformGizmo#1]
Hi Monk,

Your work is fabulous. I am using your package in our software. We will definitely give you credit once work is done.

I need one help. Currently, the gizmos appearing at target.position. Instead, I want gizmo to appear here:
targetRenderer.bounds.center
if renderer exists.

Can you please help us customise it. Many thanks!

**Comment by neuro-tech** [comment]
Hi Monk,

We have run time inspector and hierarchy in our tool. Having one extra game object is causing us a lot of trouble and forcing us to do lot of case handling.

Is there any other alternative in your mind?

**Comment by HiddenMonk** [comment]
I dont know how to Scale based off a point instead of the transforms pivot. If we can figure that out then I should be able to avoid using gameobjects.
Unfortunately, the attempt I made was not perfect (spent all day on it too T.T )
Here is the attempted code, maybe you can figure out the issues =/

```
	ScaleAround(transform, render.bounds.center, scale);

	public void ScaleAround(Transform target, Vector3 worldPivot, Vector3 newScale)
	{
		//Seemed to work, except when under a parent that has a non uniform scale and rotation it was a bit off.
		//This might be due to transform.lossyScale not being accurate under those conditions, or possibly something else is wrong...
		//Maybe things can work if we can find a way to convert the "newPosition = ..." line to use Matrix4x4 for possibly more scale accuracy.
		//However, I have tried and tried and have no idea how to do that kind of math =/

		Vector3 localOffset = target.InverseTransformPoint(worldPivot);

		Vector3 localScale = target.localScale;
		Vector3 scaleRatio = new Vector3(SafeDivide(newScale.x, localScale.x), SafeDivide(newScale.y, localScale.y), SafeDivide(newScale.z, localScale.z));
		Vector3 scaledLocalOffset = localOffset;
		scaledLocalOffset.Scale(scaleRatio);
		Vector3 newPosition = target.rotation * Vector3.Scale(localOffset - scaledLocalOffset, target.lossyScale) + target.position;

		target.localScale = newScale;
		target.position = newPosition;
	}

	float SafeDivide(float value, float divider)
	{
		if(divider == 0) return 0;
		return value / divider;
	}
```

**Comment by HiddenMonk** [comment]
Maybe even if its a little inaccurate for skewed objects, would it still help you if I at least make it an option to use this version for handling center pivots without using gameobjects?
Check my post here on the forums to see my posts where I give an example of the accuracy, and how unity editors scale can also be inaccurate.
https://forum.unity.com/threads/scale-from-a-point-other-than-pivot-almost-working-code.544302/#post-3591286

**Comment by neuro-tech** [comment]
Yes, absolutely Monk.

Thanks for all these efforts. Please share the changed files, because we have made our own changes.

**Comment by HiddenMonk** [comment]
Good news is that I have the code to replace using temporary parent gameobjects as a pivot with the same accuracy.
Bad news is that it seems using a parent gameobject as a pivot, even when placed at the center of the desired object, does not give the same result as unitys Center mode.
This leaves us wondering what unity is doing differently in Center mode.
Either way, I am going to completely remove the temporary gameobject since its no longer needed, as well as put 2 Scale types that people can switch between.
One scale type called "ScaleFrom" being like using a parent gameobject as a pivot, and one scale type called "ScaleFromOffset" being like unitys Center mode (though a bit inaccurate with skewed objects).

I talk about this more in this thread https://forum.unity.com/threads/scale-from-a-point-other-than-pivot-almost-working-code.544302/#post-3591286


There may also be some differences between unity and how I have things set up currently in regards to Center pivots,
such as unity seems to use colliders first to find how much weight the object has or something to decide how much it effects the center,
but for now we only look at the Renderer.bounds.center, so expect some differences between unity.


I have also ran into a unity bug (Unity 2017.2.0f3 (64-bit)) where when rotating a object (it was skewed, not sure if that matters) it will sometimes not have its mesh collider updated or something, causing me to not be able to click it again in certain spots until I disable and reenable the mesh collider in the inspector.
Reenabling the mesh collider also updated where unity saw the center of the object to be.
Maybe related to this bug? https://issuetracker.unity3d.com/issues/colliders-are-not-updating-to-match-attached-gameobject-location


Changes made to files - TransformGizmo, Enums, AxisInfo, ExtMathf (also moved ExtMathf from folder "Objects" to folder "Helpers")
Added new file - ExtTransform


Cross fingers no bugs =)

**Comment by neuro-tech** [comment]
Awesome. Thanks Monk!
